util/chunk/chunk.go

```diff
@@ -706,3 +706,26 @@ func (c *Chunk) ToString(ft []*types.FieldType) string {
 	}
 	return string(buf)
 }
+
+// AppendRows appends multiple rows  to the chunk.
+func (c *Chunk) AppendRows(rows []Row) {
+	c.AppendPartialRows(rows)
+	c.numVirtualRows += len(rows)
+}
+
+// AppendPartialRows appends multiple row to the chunk.
+func (c *Chunk) AppendPartialRows(rows []Row) {
+	for i, dstCol := range c.columns {
+		for _, srcRow := range rows {
+			appendCellByCell(dstCol, srcRow.c.columns[i], srcRow.idx)
+		}
+	}
+}
+
+// LeftRequiredRows  chunk may append the number of rows
+func (c *Chunk) LeftRequiredRows(offset int) int {
```

Comment:
I think return require rows directly is better. Appending how many rows should be decided by the caller.

Reply:
the code will have many call.